packages/serverless/src/domain/use-case/merkle-tree.ts

```diff
@@ -1,166 +1,174 @@
-import { ModelMerkleTree } from '@zkdb/storage';
-import { ClientSession } from 'mongodb';
-import ModelMetadataDocument from '../../model/database/metadata-document.js';
-import { Field } from 'o1js';
 import {
-  TMerkleNode,
+  TMerkleNodeDetailJson,
+  TMerkleNodeJson,
   TMerkleProof,
   TMerkleTreeInfo,
-  TPaginationReturn,
   TPagination,
-  TMerkleJson,
+  TPaginationReturn,
 } from '@zkdb/common';
+import { ModelMerkleTree } from '@zkdb/storage';
+import { ClientSession } from 'mongodb';
+import { Field } from 'o1js';
+import ModelMetadataDocument from '../../model/database/metadata-document.js';
 
-export async function getWitnessByDocumentId(
-  databaseName: string,
-  docId: string,
-  session?: ClientSession
-): Promise<TMerkleProof[]> {
-  const modelDocumentMetadata = new ModelMetadataDocument(databaseName);
-
-  const docMetadata = await modelDocumentMetadata.findOne({
-    docId,
-  });
-
-  if (!docMetadata) {
-    throw Error(`Metadata has not been found`);
+export class MerkleTree {
+  public static async document(
+    databaseName: string,
+    docId: string,
+    session?: ClientSession
+  ): Promise<TMerkleProof[]> {
+    const modelDocumentMetadata = new ModelMetadataDocument(databaseName);
+
+    const docMetadata = await modelDocumentMetadata.findOne({
+      docId,
+    });
```

Comment:
You should put session in `findOne`

Reply:
Can't use find one, since we need latest document